### DISCUSSION TITLE
All uploads stuck in "upload". Library folder is empty.

### DISCUSSION THREAD

**Discussion by strangeh21** [immich-app/immich#12092] [Q&A] [answered]
Hi there!
I'm running into an issue. 
I had not updated Immich in a while, and then when I did update it broke the entire install so I did a fresh install. 
I took all photos from the old immich install and imported them all using [immich-go](https://github.com/simulot/immich-go). 
Everything seems to work just fine on immich side of things. There was a few untracked and 5 offline files. I thought nothing of it. 
Based on other threads I've read, it'll magically solve itself, so I checked there were no errors in the docker compose logs, and then I left it. 
It's been a month, and my library folder is still empty, my upload folder is growing. But all pictures are working, and uploaded from my phone correctly.

Only "special" thing about the setup is it's a VM running on Proxmox, and that I mounted a 8TB volume from a 40TB RAID6 partition (With 24TB usable space) where the volumes are stored. Shouldn't cause any issues, it's mounted locally, directly to the VM, no SMB or CIFS setup. 
```
$ df -h
Filesystem      Size  Used Avail Use% Mounted on
udev            5.9G     0  5.9G   0% /dev
tmpfs           1.2G   11M  1.2G   1% /run
/dev/sdb1       125G   48G   71G  41% /
tmpfs           5.9G     0  5.9G   0% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
/dev/sdc1       7.8T  1.5T  6.0T  20% /mnt/hdd/other
/dev/sda1        16T   13T  2.3T  85% /mnt/hdd/media
tmpfs           1.2G     0  1.2G   0% /run/user/0
tmpfs           1.2G     0  1.2G   0% /run/user/1000

$ lsblk
NAME   MAJ:MIN RM  SIZE RO TYPE MOUNTPOINTS
sda      8:0    0 15.6T  0 disk 
└─sda1   8:1    0 15.6T  0 part /mnt/hdd/media
sdb      8:16   0  128G  0 disk 
├─sdb1   8:17   0  127G  0 part /
├─sdb2   8:18   0    1K  0 part 
└─sdb5   8:21   0  975M  0 part 
sdc      8:32   0  7.8T  0 disk 
└─sdc1   8:33   0  7.8T  0 part /mnt/hdd/other
sr0     11:0    1 1024M  0 rom  


$ sudo fdisk -l
Disk /dev/sda: 15.63 TiB, 17179869184000 bytes, 33554432000 sectors
Disk model: QEMU HARDDISK   
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: AF5AFCD0-705F-4725-9C68-1F0882FD9841

Device     Start         End     Sectors  Size Type
/dev/sda1   2048 33554429951 33554427904 15.6T Linux filesystem


# THIS IS THE PARTITION IMMICH IS STORING PICTURES
Disk /dev/sdc: 7.81 TiB, 8589934592000 bytes, 16777216000 sectors
Disk model: QEMU HARDDISK   
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: B3FBBC6C-602A-4613-9B0D-D5527EBF8968

Device     Start         End     Sectors  Size Type
/dev/sdc1   2048 16777213951 16777211904  7.8T Linux filesystem

# THIS IS THE BOOT PARTITION WHERE DOCKER IS RUNNING FROM
Disk /dev/sdb: 128 GiB, 137438953472 bytes, 268435456 sectors
Disk model: QEMU HARDDISK   
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: dos
Disk identifier: 0xcb25fb58

Device     Boot     Start       End   Sectors  Size Id Type
/dev/sdb1  *         2048 266434559 266432512  127G 83 Linux
/dev/sdb2       266436606 268433407   1996802  975M  5 Extended
/dev/sdb5       266436608 268433407   1996800  975M 82 Linux swap / Solaris
```

I'd love to know what's wrong, or at the very least where I can find out what's wrong. 

docker-compose.yml

```dockerfile
# WARNING: Make sure to use the docker-compose.yml of the current release:
#
# https://github.com/immich-app/immich/releases/latest/download/docker-compose.yml
#
# The compose file on main may not be compatible with the latest release.
#

name: immich

services:
  immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    # extends:
    #   file: hwaccel.transcoding.yml
    #   service: cpu # set to one of [nvenc, quicksync, rkmpp, vaapi, vaapi-wsl] for accelerated transcoding
    volumes:
      # Do not edit the next line. If you want to change the media storage location on your system, edit the value of UPLOAD_LOCATION in the .en>
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
    env_file:
      - .env
    ports:
      - 2283:3001
    depends_on:
      - redis
      - database
    restart: always

  immich-machine-learning:
    container_name: immich_machine_learning
    # For hardware acceleration, add one of -[armnn, cuda, openvino] to the image tag.
    # Example tag: ${IMMICH_VERSION:-release}-cuda
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}
    # extends: # uncomment this section for hardware acceleration - see https://immich.app/docs/features/ml-hardware-acceleration
    #   file: hwaccel.ml.yml
    #   service: cpu # set to one of [armnn, cuda, openvino, openvino-wsl] for accelerated inference - use the `-wsl` version for WSL2 where app>
    volumes:
      - model-cache:/cache
    env_file:
      - .env
    restart: always

  redis:
    container_name: immich_redis
    image: docker.io/redis:6.2-alpine@sha256:e3b17ba9479deec4b7d1eeec1548a253acc5374d68d3b27937fcfe4df8d18c7e
    healthcheck:
      test: redis-cli ping || exit 1
    restart: always

  database:
    container_name: immich_postgres
    image: docker.io/tensorchord/pgvecto-rs:pg14-v0.2.0@sha256:90724186f0a3517cf6914295b5ab410db9ce23190a2d9d0b9dd6463e3fa298f0
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
      POSTGRES_INITDB_ARGS: '--data-checksums'
    volumes:
      # Do not edit the next line. If you want to change the database storage location on your system, edit the value of DB_DATA_LOCATION in the>
      - ${DB_DATA_LOCATION}:/var/lib/postgresql/data
    healthcheck:
      test: pg_isready --dbname='${DB_DATABASE_NAME}' --username='${DB_USERNAME}' || exit 1; Chksum="$$(psql --dbname='${DB_DATABASE_NAME}' --us>
      interval: 5m
      start_interval: 30s
      start_period: 5m
    command: ["postgres", "-c" ,"shared_preload_libraries=vectors.so", "-c", 'search_path="$$user", public, vectors', "-c", "logging_collector=o>
    restart: always

volumes:
  model-cache:
``` 

.env
```
# You can find documentation for all the supported env variables at https://immich.app/docs/install/environment-variables

# The location where your uploaded files are stored
UPLOAD_LOCATION=/mnt/hdd/other/immich
# The location where your database files are stored
DB_DATA_LOCATION=./postgres

# To set a timezone, uncomment the next line and change Etc/UTC to a TZ identifier from this list: 
# https://en.wikipedia.org/wiki/List_of_tz_database_time_zones#List
TZ=Europe/Bucharest

# The Immich version to use. You can pin this to a specific version like "v1.71.0"
IMMICH_VERSION=release

# Connection secret for postgres. You should change it to a random password
DB_PASSWORD=xxxxxxxxx

# The values below this line do not need to be changed
###################################################################################
DB_USERNAME=postgres
DB_DATABASE_NAME=immich

``` 

How folders look (Free space is incorrect):
![image](https://github.com/user-attachments/assets/9e1c2cd9-73ac-4439-89bb-198c0da88c36)

Repair page on Immich: 
![image](https://github.com/user-attachments/assets/56245af4-1543-49f2-b9ad-0c3148041e05)

docker compose logs: [logs.txt](https://github.com/user-attachments/files/16782089/logs.txt)
I see some errors regarding thumbnail generation, if that's the thing causing the issue, how would I systematically go through all potential "problematic" files, or move them all off the server, and go through them manually?

Finally, permissions: 
```
/docker/immich$ ls -al
total 20
drwxr-xr-x  3 mediaserver mediaserver 4096 Aug 28 06:35 .
drwxr-xr-x  6 root        root        4096 Jul 15 01:46 ..
-rw-r--r--  1 mediaserver mediaserver 3126 Jul 30 15:19 docker-compose.yml
-rw-r--r--  1 mediaserver mediaserver  897 Aug  1 02:18 .env
drwx------ 21         999 root        4096 Aug 27 20:00 postgres

/mnt/hdd/other/immich$ ls -al
total 24
drwxrwxr-x 6 mediaserver share       4096 Aug  1 05:16 .
drwxrwxr-x 7 mediaserver mediaserver 4096 Aug  1 06:11 ..
drwxr-xr-x 4 root        root        4096 Aug  6 06:51 encoded-video
drwxr-xr-x 2 root        root        4096 Aug  1 04:14 library
drwxr-xr-x 4 root        root        4096 Aug  6 06:50 thumbs
drwxr-xr-x 4 root        root        4096 Aug  6 06:50 upload

```

**Comment by JW-CH** [comment, marked as answer]
If storage template (https://immich.app/docs/administration/storage-template/) is not enabled, all of your media gets stored in the `UPLOAD_LOCATION` folder.

See here for more information: https://immich.app/docs/administration/backup-and-restore/#asset-types-and-storage-locations
> The UPLOAD_LOCATION/library folder is not used by default on new machines running version 1.92.0. It is used only if the system administrator activated the storage template engine, for more info read the [release notes](https://github.com/immich-app/immich/releases/tag/v1.92.0#:~:text=the%20partner%E2%80%99s%20assets.-,Hardening%20storage%20template).

> If you choose to activate the storage template engine, it will move all assets to UPLOAD_LOCATION/library/<userID>

**Reply by strangeh21** [reply]
Oh. I did not know that. Thank you. I guess that's the problem coming from a MUCH earlier version, miss things like this!

If I ever need to shutdown and start a fresh immich instance or (god forbid) migrate to a new app, is it easier to manage with storage template enabled? 
As far as I can tell, the upload folder is just randomly assigned variables, virtually impossible to manage outside of this current install.

**Reply by JW-CH** [reply]
In my opinion, yes. It is easier. The upload folder is a mess (intended) and the storage template does exactly what it should. I am quite happy with `{{y}}/{{y}}-{{MM}}-{{dd}}/{{filename}}` as my template. Note that you are able to configure it to your likings. Hope that helps!

**Reply by strangeh21** [reply]
Thank you very much for the help! Yes, although personally I have always preferred `{{y}}/{{y}}-{{MM}}/{{y}}-{{MM}}-{{dd}}/{{filename}}`
If it were possible I'd also add timestamp to the filename, but since I cannot guarantee timestamps are always correct, then it seems pointless. 

As a side thought, if I change the timestamp of a picture inside of immich, will that picture be moved to the correct folder within library? 
As in, I have some pictures that are all set to 1999-01-01, and I am going to slowly go through them to set the correct date. 
Will those automatically move to the correct library folder after updating the date?

**Reply by JW-CH** [reply]
I assume that this will be the case. Maybe @bo0tzz could confirm?

About the filename stuff. Yes, you can add the date to your filename. Just do something like this: `{{y}}/{{y}}-{{MM}}/{{y}}-{{MM}}-{{dd}}/{{y}}-{{MM}}-{{dd}}_{{filename}}`. If you edit your template [here](https://my.immich.app/admin/system-settings?isOpen=storage-template) there is a nice preview what the outcome would be.

**Reply by bo0tzz** [reply]
> if I change the timestamp of a picture inside of immich, will that picture be moved to the correct folder within library?

I'm not sure whether the template migration job triggers automatically in this case, but whenever that runs the file will be moved yes.

**Reply by strangeh21** [reply]
I had assumed that would break something down the line, because of potential recursive changes.
If it is: `{{y}}-{{MM}}-{{dd}}_{{filename}}`:
1. `2024-08-28_IMG1234.jpg`
2. `2024-08-28_2024-08-28_IMG1234.jpg`
3. `2024-08-28_2024-08-28_2024-08-28_IMG1234.jpg`
etc....etc...

As in, first iteration will be `date_filename`, but then filename is now `date_filename`, so if it changes in the future, then it'll apply a date to the new filename, thus adding date to `date_filename` which would then end with `date_date_filename`.

Maybe that's not the case, but I can see a scenario in the next 5-10 years that would require a full reinstall, and full migration to a fresh install of immich ~v200 or so, where the now filename, includes the date, and causes the above issue. 

I think it's best for me to stick to not having the date in the filename, just in case :)

**Reply by bo0tzz** [reply]
> As in, first iteration will be date_filename, but then filename is now date_filename, so if it changes in the future, then it'll apply a date to the new filename, thus adding date to date_filename which would then end with date_date_filename

The filename that is stored in the DB is always the original one from when the file was first uploaded. As long as you have that you're good. If you lose the DB and are uploading to a new instance, the scenario you mention could happen (at a depth of 1).

**Comment by Blackspell01** [comment]
I have the same issue but im not sure if this is an intended behaviour or not. I have set up a storage template and restarted the container but the files are still ending up in /upload/UID/... . Only after I hit the start migration task button manually the files are copied in the /library/user/date/.. path correctly. Is this normal?

**Reply by bo0tzz** [reply]
Yes that's normal.
